typescript/sdk/src/core/HyperlaneRelayer.ts

```diff
@@ -98,12 +104,18 @@ export class HyperlaneRelayer {
   async getIsmConfig(
     chain: ChainName,
     ism: Address,
+    messageContext?: DispatchedMessage,
   ): Promise<DerivedIsmConfig> {
     let config: DerivedIsmConfig | undefined;
     if (this.cache?.ism[chain]?.[ism]) {
       config = this.cache.ism[chain][ism] as DerivedIsmConfig | undefined;
     } else {
-      const evmIsmReader = new EvmIsmReader(this.multiProvider, chain);
+      const evmIsmReader = new EvmIsmReader(
+        this.multiProvider,
+        chain,
+        undefined,
+        messageContext,
```

Comment:
why does it need to be initialized with this?

Reply:
https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4765#discussion_r1819179492